### PR TITLE
Joe McNally Bowling Challenge JS Submission

### PR DESCRIPTION
**Pull request by jmcnally17**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,37 +2,38 @@
...
+then begin entering the rolls for each frame. This will give you an array of all frames together as well as a total score for the game that is submitted.
+
+### Tests
+
+In order to run the tests for each class, in the [main](https://github.com/jmcnally17/bowling-challenge-javascript) directory, simply run:
+```
+jest
+```
+These tests in the [test](https://github.com/jmcnally17/bowling-challenge-javascript/tree/main/test) folder were written using the TDD process and cover each function present in the classes including their edge cases.
+
```
Comment (baljitrakhra) [inline README.md:35]: I really liked your readme. It is vert succinct but with all the information

File: lib/frame.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    return this.pinsRemaining === 0 && this.firstRoll != 10
+  }
+
+  isStrike() {
+    return this.firstRoll === 10
+  }
+
+  // private functions
+
+  #privateCheckError(num) {
```
Comment (baljitrakhra) [inline lib/frame.js:40]: I really liked this error checker

File: lib/scorecardInterface.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+console.log("Welcome to this bowling scorecard program!");
+console.log("Please begin entering your rolls.");
+
+const rl = readLine.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+const card = new Scorecard;
```
Comment (baljitrakhra) [inline lib/scorecardInterface.js:15]: I think this may be a good syntax : 
const card = new Scorecard();

File: lib/scorecardInterface.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+let count = 0;
+
+const recursiveAsyncReadLine = function () {
+  if (card.frames.length < 10) {
```
Comment (baljitrakhra) [inline lib/scorecardInterface.js:19]: This need to be broken down in few functions and made DRY and DRE

File: test/scorecard.test.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
+const Scorecard = require('../lib/scorecard');
+
```
Comment (baljitrakhra) [inline test/scorecard.test.js:2]: I really enjoyed reading the mock. It is very comprehensive suit of BDD testing

**Review by baljitrakhra — COMMENTED** [review]
I really enjoyed reading the code. The use of mocking is brilliant.
